packages/extension-core/src/domains/balances/pool.ts

```diff
@@ -0,0 +1,865 @@
+            error?.type === "CHAIN_CONNECTION_ERROR"
+          ) {
+            const addressesByModuleToken = subscriptionParameters[balanceModule.type] ?? {}
+            // set status to stale for balances matching the error
+            const staleObservable = this.#pool.pipe(
+              map((val) =>
+                Object.values(val)
+                  .filter(({ tokenId, address, source, ...rest }) => {
+                    const locationId = "chainId" in rest ? rest.chainId : rest.evmNetworkId
+                    const chainComparison = error.chainId
+                      ? error.chainId === locationId
+                      : error.evmNetworkId
+                      ? error.evmNetworkId === locationId
+                      : true
+                    return (
+                      chainComparison &&
+                      addressesByModuleToken[tokenId]?.includes(address) &&
+                      source === balanceModule.type
+                    )
+                  })
+                  .map((balance) => ({ ...balance, status: "stale" } as BalanceJson))
+              )
+            )
+
+            firstValueFrom(staleObservable).then((v) => {
+              if (v.length) this.updatePool(v)
+            })
+          } else if (error) {
+            log.error("Balances Pool unknown error", error)
+          }
+
+          // good balances
+          if (result) {
+            if ("status" in result) {
+              // For modules using the new SubscriptionResultWithStatus pattern
+              if (result.status === "initialising") this.#initialising.add(balanceModule.type)
+              else this.#initialising.delete(balanceModule.type)
+              this.updatePool(result.data)
+            } else {
+              // add good ones to initialisedBalances
+              this.updatePool(Object.values(result.toJSON()))
+            }
+          }
+        }
+      )
+    )
+
+    this.#closeSubscriptionCallbacks = this.#closeSubscriptionCallbacks.concat(
+      closeSubscriptionCallbacks
+    )
+  }
+
+  /**
+   * Closes all balance subscriptions.
+   */
+  private async closeSubscriptions() {
+    if (this.#subscriptionsState === "Closing")
+      await firstValueFrom(this.#subscriptionsStateUpdated)
+
+    if (this.#subscriptionsState !== "Open") return
+    this.setSubscriptionsState("Closing")
+    log.log("Closing balance subscriptions")
+
+    // ignore old subscriptions if they're still closing when we next call `openSubscriptions()`
+    this.#subscriptionsGeneration = (this.#subscriptionsGeneration + 1) % Number.MAX_SAFE_INTEGER
+
+    this.#closeSubscriptionCallbacks
+      .splice(0, this.#closeSubscriptionCallbacks.length)
+      // wait 10_000ms in case the user is opening and closing the popup quickly
+      // this way the rpcs will remain connected for an extra ten seconds
+      .forEach((cb) => cb.then((close) => setTimeout(close, 10_000)))
```

Comment:
@alecdwm wdyt about putting the `this.setSubscriptionsState("Closed")` in a `then` following this timeout.... that would ensure the subscriptions were actually closed when the balance modules wind down

Reply:
I think it should stay where it is;
`setSubscriptionsState` is used to prevent two *active* sets of balance subscriptions from running at the same time.

By that, I mean that the effect of having the state be anything other than `Closed` is just that we're prevented from opening any new subscriptions.

It should have no effect on whether or not we manage to actually close the existing subscriptions when we ask to

Reply:
The mechanism for killing off the existing subscriptions is the `subscriptionsGeneration` key. But we could replace that with an `AbortController` - which would make the code a bit tidier.